src/main/java/backend/resource/Model.java

```diff
@@ -211,6 +211,22 @@ private void ______OPERATIONS_____() {}
                 () -> logger.error("Issue " + issueId + " not found in model for " + repoId));
     }
 
+    /**
+     * Replaces the milestone of an issue specified by {@code issueId} with {@code milestone}
+     * @param issueId
+     * @param milestone
+     * @return the modified TurboIssue if successful
+     */
+    public synchronized Optional<TurboIssue> replaceIssueMilestone(int issueId, Integer milestone) {
+        Optional<TurboIssue> issueLookUpResult = getIssueById(issueId);
+        return Utility.safeFlatMapOptional(issueLookUpResult,
```

Comment:
Actually, now that I'm in the midst of trying to change this, I realised that a lot of things will need to be changed, including label change logic flow since it will affect shared methods (e.g. Logic's getIssue). I think this should be done in a separate PR.